@xen-orchestra/lite/scripts/release.mjs

```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env zx
+
+import argv from "minimist";
+import { tmpdir } from "os";
+
+$.verbose = false;
+
+const DEPLOY_SERVER = "www-xo.gpn.vates.fr";
+
+const { version: pkgVersion } = await fs.readJson("./package.json");
+
+const opts = argv(process.argv, {
+  boolean: ["help", "build", "deploy", "ghRelease", "tarball"],
+  string: [
+    "base",
+    "dist",
+    "ghToken",
+    "tarballDest",
+    "tarballName",
+    "username",
+    "version",
+  ],
+  alias: {
+    u: "username",
+    h: "help",
+    "gh-release": "ghRelease",
+    "gh-token": "ghToken",
+    "tarball-dest": "tarballDest",
+    "tarball-name": "tarballName",
+  },
+  default: {
+    dist: "dist",
+    version: pkgVersion,
+  },
+});
+
+let {
+  base,
+  build,
+  deploy,
+  dist,
+  ghRelease,
+  ghToken,
+  help,
+  tarball,
+  tarballDest,
+  tarballName,
+  username,
+  version,
+} = opts;
+
+const usage = () => {
+  console.log(
+    `Usage: ./release.mjs
+    [--help|-h - show this message]
+
+    [--version X.Y.Z - XO Lite version - default: package.json version]
+    [--dist /path/to/folder - build destination folder - default: dist]
+
+    [
+      --build - whether to build XO Lite or not
+      [--base url - base URL for assets - default: "/" or "lite.xen-orchestra.com/dist" if --deploy is passed]
+    ]
+
+    [
+      --tarball - whether to generate a tarball or not
+      [--tarball-dest /path/to/folder - tarball destination folder]
+      [--tarball-name file.tar.gz - tarball file name - default xo-lite-X.Y.Z.tar.gz]
+    ]
+
+    [
+      --gh-release - whether to release on GitHub or not
+      [--gh-token token - GitHub API token with "Contents" write permissions]
+    ]
+
+    [
+      --deploy - whether to deploy to xen-orchestra.com or not
+      --username|-u <LDAP username>
+    ]
+`
+  );
+};
+
+if (help) {
+  usage();
+  process.exit(0);
+}
+
+const yes = async (q) =>
+  ["y", "yes"].includes((await question(q + " [y/N] ")).toLowerCase());
+
+const no = async (q) => !(await yes(q));
+
+const step = (s) => console.log(chalk.green.bold(`\n${s}\n`));
+
+const stop = () => {
+  console.log(chalk.yellow("Stopping"));
+  process.exit(0);
+};
+
+const ghApiCall = async (path, method = "GET", data) => {
+  const opts = {
+    method,
+    headers: {
+      Accept: "application/vnd.github+json",
+      Authorization: `Bearer ${ghToken}`,
+      "X-GitHub-Api-Version": "2022-11-28",
+    },
+  };
+
+  if (data !== undefined) {
+    opts.body = typeof data === "object" ? JSON.stringify(data) : data;
+  }
+
+  const res = await fetch(
+    "https://api.github.com/repos/vatesfr/xen-orchestra" + path,
+    opts
+  );
+
+  if (res.status === 404 || res.status === 422) {
+    return;
+  }
```

Comment:
If I'm not mistaken, 404 and 422 are never used by GitHub's API for errors but only to say "the object doesn't exist". So if we get an error before or after that, we'll see it, but if we fall in this exact case, I'm returning `undefined` so that it's easier to handle in the code, since an object not being found is not considered to be an error. (e.g.: the release doesn't exist)